File: core/core-api/src/test/java/com/coffee/gu/payment/ShedlockEntity.java
package com.coffee.gu.payment;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

@Entity
@Table(name = "shedlock")
public class ShedlockEntity {

    @Id
    @Column(name = "name", length = 64, nullable = false)
    private String name;

    @Column(name = "lock_until", nullable = false)
    private Instant lockUntil;

    @Column(name = "locked_at", nullable = false)
    private Instant lockedAt;

    @Column(name = "locked_by", nullable = false)
    private String lockedBy;

    public ShedlockEntity() {
    }

    public ShedlockEntity(String name, Instant lockUntil, Instant lockedAt, String lockedBy) {
        this.name = name;
        this.lockUntil = lockUntil;
        this.lockedAt = lockedAt;
        this.lockedBy = lockedBy;
    }

    public String getName() {
        return name;
    }

    public Instant getLockUntil() {
        return lockUntil;
    }

    public Instant getLockedAt() {
        return lockedAt;
    }

    public String getLockedBy() {
        return lockedBy;
    }
}

File: core/core-api/src/test/kotlin/com/coffee/gu/payment/ShedlockEntity.kt
package com.coffee.gu.payment

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant

@Entity
@Table(name = "shedlock")
class ShedlockEntity(
    @Id
    @Column(name = "name", length = 64, nullable = false)
    val name: String = "",

    @Column(name = "lock_until", nullable = false)
    val lockUntil: Instant = Instant.now(),

    @Column(name = "locked_at", nullable = false)
    val lockedAt: Instant = Instant.now(),

    @Column(name = "locked_by", nullable = false)
    val lockedBy: String = ""
)
